base/setup/usetup/usetup.c

```diff
@@ -4532,8 +4532,14 @@ ProgressTimeOutStringHandler(
     OUT PSTR Buffer,
     IN SIZE_T cchBufferSize)
 {
+    static ULONG BarWidth = 0;
```

Comment:
why is this static?

Reply:
The idea as far as I can see, is that the BarWidth is not recomputed everytime this function is called.

Reply:
To calculate it once, the variable value should stay in memory with subsequent procedure calls.

Reply:
I'm wondering whether it would have been more efficient if doing instead:
```c
static ULONG BarWidth = max(1, Bar->Width);
```

Reply:
No, it won't work that way:
```
../base/setup/usetup/usetup.c: In function 'ProgressTimeOutStringHandler':
../base/setup/usetup/usetup.c:4535:5: error: initializer element is not constant
```

Reply:
You're worried about recalculating a max() operation?
The extra lines of code and loss of readability created by this outweigh the performance benefit by miles.

Reply:
Yeah this seems like a nonsense 'optimalization'.